File: 18_Bit_Manipultion/03_Count_bits/main.py
class Solution:
    def countBits(self, n: int) -> list[int]:
        ans = [0] * (n + 1) # Initialize the result list with size n+1
        
        for i in range(1, n + 1):
            ans[i] = ans[i >> 1] + (i & 1) # Calculate the number of 1s
        
        return ans

File: 18_Bit_Manipultion/03_Count_bits/main.cpp

#include <vector>
using namespace std;

class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans(n + 1, 0); // Initialize the result vector with size n+1
        
        for (int i = 1; i <= n; ++i) {
            ans[i] = ans[i >> 1] + (i & 1); // Calculate the number of 1s
        }
        
        return ans;
    }
};
